var/spack/repos/builtin/packages/libzmq/package.py

```diff
@@ -57,6 +57,8 @@ class Libzmq(AutotoolsPackage):
 
     # Fix aggressive compiler warning false positive
     patch('https://github.com/zeromq/libzmq/commit/92b2c38a2c51a1942a380c7ee08147f7b1ca6845.patch?full_index=1', sha256='310b8aa57a8ea77b7ac74debb3bf928cbafdef5e7ca35beaac5d9c61c7edd239', when='@4.2.3:4.3.4 %gcc@11:')
+    # Gcc12 patch fix
+    patch('https://github.com/zeromq/libzmq/commit/a01d259db372bff5e049aa966da4efce7259af67.patch?full_index=1', sha256='edca864cba914481a5c97d2e975ba64ca1d2fbfc0044e9a78c48f1f7b2bedb6f', when='@:4.3.4 %gcc@12:')
```

Comment:
One caveat. This patch as written should apply to all versions we currently support, did you check that? Otherwise it would be better to restrict the versions where the patch applies and add a conflict for earlier versions and GCC 12

Reply:
No I didn't check for this.  what would be the magic to put in a conflict?

Reply:
This:
```python
when='@:4.3.4 %gcc@12:'
```
means "when GCC 12 or greater is used and the version is 4.3.4 or lower", you just need to be more specific on the version if the patch doesn't apply to v4.0 or v4.1 etc.

Reply:
@joequant